src/openrct2/world/park.c

```diff
@@ -477,13 +477,17 @@ static sint32 park_calculate_guest_generation_probability()
 	return probability;
 }
 
+/**
+ * Choose a random peep spawn and iterates through until defined spawn is found.
+ */
 static void get_random_peep_spawn(rct2_peep_spawn *spawn)
 {
 	*spawn = gPeepSpawns[0];
-	if (gPeepSpawns[1].x != PEEP_SPAWN_UNDEFINED) {
-		if (scenario_rand() & 0x80000) {
-			*spawn = gPeepSpawns[1];
-		}
+	rct2_peep_spawn spawnList[MAX_PEEP_SPAWNS];
+	uint32 numSpawns = map_get_available_peep_spawn_list(spawnList);
+	if (numSpawns > 0) {
+		uint32 index = scenario_rand() % numSpawns;
+		*spawn = spawnList[index];
 	}
 }
```

Comment:
huh? i'm a little confused, could you clear it up a little? have the "get_random_peep_spawn" return an index? or have map_get_available_peep_spawn_list return indexes?